pkg/providers/aws/cluster_network_provider.go

```diff
@@ -1607,6 +1619,38 @@ func (n *NetworkProvider) getNonOverlappingDefaultCIDR(ctx context.Context) (*ne
 	return nil, errorUtil.New("could not find a default cidr block")
 }
 
+func (n *NetworkProvider) getPodCIDR(ctx context.Context, logger *logrus.Entry) (*net.IPNet, error) {
+	networkConf := &v12.Network{}
+	err := n.Client.Get(ctx, client.ObjectKey{Name: "cluster"}, networkConf)
+	if err != nil {
+		return nil, errorUtil.Wrap(err, "failed to get network kind")
+	}
+	var podCIDR *net.IPNet
+	for _, entry := range networkConf.Spec.ClusterNetwork {
+		_, podCIDR, err = net.ParseCIDR(entry.CIDR)
+		if err != nil {
+			panic(err)
```

Comment:
Not a good idea to panic, best to return the error and let it be handled at the controller level. 
```suggestion
			return nil, errorUtil.Wrap(err, "error parsing pod cidr")
```

Reply:
yup was meant to change that before i un draft the pr good catch